lib/Config.py

```diff
@@ -175,18 +175,28 @@ def args(self, value):
 
     def check_media_dir(self, stages):
         if self._args.get("media_dir", None) is None:
-            # if `--media-dir` was not specified on the commandline, default
-            # it to self.media_base_dir/self.activity.name
-            self._args["media_dir"] = os.path.join(self.media_base_dir, self.activity.name)
 
+            # Ensure the activity is initialized.  This will result in the
+            # activity dictionary being read, and possibly getting the last
+            # media_dir used.
+            self.activity.name
```

Comment:
Is there a better way to do this?

Reply:
You could have the activity set media_dir from the activity file as a property instead of shoving it into args then just using next to replace most of the if/else code entirely

`default_media_dir = os.path.join(self.media_base_dir, self.activity.name)

media_dir = next((el for el in [self._args.get("media_dir",None), self.activity.media_dir, default_media_dir] if el is not None), None)
self._args["media_dir"] = media_dir

for adir in [media_dir, os.path.abspath(media_dir), os.path.join(self.media_base_dir, media_dir)]:
...
`